src/state/translations-mixin.ts

```diff
@@ -221,9 +221,18 @@ export default <T extends Constructor<HassBaseEl>>(superClass: T) =>
       // loaded resources. This merges the new data on top of the old data for
       // this language, so that the full translation set can be loaded across
       // multiple fragments.
+      //
+      // Beware of a subtle race condition: it is possible to get here twice
+      // before this.hass is even created. In this case our base state comes
+      // from this._pendingHass instead. Otherwise the first set of strings is
+      // overwritten when we call _updateHass the second time!
+      const baseHass = this.hass ?? this._pendingHass;
       const resources = {
+        // Spread any other language resources before adding the new one to
+        // avoid possible race conditions related to language switching.
+        ...baseHass?.resources,
```

Comment:
No idea if there *is* such a race condition, but "hey let's throw away any other language data and assume `language` is the current language" seemed suspicious to me. Best to be safe, since there's no real cost to it here.

Reply:
Let's move the check of line 240 up, so don't end up with all languages in memory, I don't think we need to update the resources if the language of the resources doesn't match the current language?

Reply:
Sure. Let me tweak this.